cmd/planner/main.go

```diff
@@ -73,18 +74,66 @@ func main() {
 		log.Fatalln("failed to connect to db:", err)
 	}
 
+	// Start AMQP consumer
 	consumer := queue.NewAmqpConsumer(rabbitMQUri, queueName, db, nil)
-	consumerCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	var consumerWg sync.WaitGroup
+	consumerWg.Add(1)
 	go func() {
-		if err := consumer.Start(consumerCtx); err != nil {
-			log.Fatalf("failed to start AMQP consumer: %v", err)
+		defer consumerWg.Done()
+		if err := consumer.Start(ctx); err != nil && !errors.Is(err, context.Canceled) {
+			log.Printf("AMQP consumer error: %v", err)
 		}
 	}()
 
+	// Create HTTP server
 	metrics := planner.NewMetrics(db)
-	server := planner.NewServer(db, metrics)
+	handler := planner.NewServer(db, metrics)
+	server := &http.Server{
+		Addr:    ":" + port,
+		Handler: otelhttp.NewHandler(handler, "planner-api", otelhttp.WithServerName("planner")),
+	}
+
+	// Start HTTP server
+	go func() {
+		log.Println("Starting planner API server on port", port)
+		if err := server.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			log.Fatalf("HTTP server error: %v", err)
+		}
+	}()
+
+	// Wait for shutdown signal
+	<-ctx.Done()
+	log.Println("Shutdown signal received, starting graceful shutdown...")
+
+	// Create shutdown context with timeout
+	shutdownCtx, shutdownCancel := context.WithTimeout(context.Background(), shutdownTimeout)
+	defer shutdownCancel()
+
+	// Shutdown HTTP server (stop accepting new requests)
+	log.Println("Shutting down HTTP server...")
+	if err := server.Shutdown(shutdownCtx); err != nil {
+		log.Printf("HTTP server shutdown error: %v", err)
+	}
+
+	// Wait for consumer to finish processing current message
+	log.Println("Waiting for AMQP consumer to stop...")
+	consumerWg.Wait()
+
+	// Close AMQP connection
+	log.Println("Closing AMQP connection...")
+	if err := consumer.Close(); err != nil {
+		log.Printf("failed to close AMQP consumer: %v", err)
+	}
+
+	// Close database connection pool
+	log.Println("Closing database connection...")
+	db.Close()
+
+	// Shutdown telemetry (flush metrics and traces)
+	log.Println("Shutting down telemetry...")
+	if err := telemetry.Shutdown(shutdownCtx); err != nil {
+		log.Printf("failed to shutdown telemetry: %v", err)
+	}
```

Comment:
Can we put this logic into a separate function? E.g. with name `shutdown` so we have a function with clear scope handling the details of shutdown, and if a reader of the code is interested in details, can hop to that?